src/librustdoc/html/static/js/main.js

```diff
@@ -979,6 +979,121 @@ function hideThemeButtonState() {
     onHashChange(null);
     window.addEventListener("hashchange", onHashChange);
     searchState.setup();
+
+    /////// --scrape-examples interactions
```

Comment:
I'm wondering if we should put this JS into its own file (which would be included if the feature is enabled). The file is already big enough...

What do you think @willcrichton ?

Reply:
FWIW, if the feature is disabled, then the runtime cost of this code is just querying an empty selector and then doing nothing.

But I can move it into a separate file if y'all like.

Reply:
Yes please!

Reply:
Ok I've factored it out into a different file. Please let me know if it seems correct / idiomatic.

Reply:
Looks perfect, thanks!